Common/Scheduling/DateRules.cs

```diff
@@ -108,12 +108,75 @@ public IDateRule MonthStart()
         /// Specifies an event should fire on the first tradeable date for the specified
         /// symbol of each month
         /// </summary>
-        /// <param name="symbol">The symbol whose exchange is used to determine the first 
+        /// <param name="symbol">The symbol whose exchange is used to determine the first
         /// tradeable date of the month</param>
         /// <returns>A date rule that fires on the first tradeable date for the specified security each month</returns>
         public IDateRule MonthStart(Symbol symbol)
         {
-            return new FuncDateRule(symbol.ToString() + ": MonthStart", (start, end) => MonthStartIterator(GetSecurity(symbol), start, end));
+            return new FuncDateRule(symbol.Value + ": MonthStart", (start, end) => MonthStartIterator(GetSecurity(symbol), start, end));
+        }
+
+        /// <summary>
+        /// Specifies an event should fire on the last of each month
+        /// </summary>
+        /// <returns>A date rule that fires on the last of each month</returns>
+        public IDateRule MonthEnd()
+        {
+            return new FuncDateRule("MonthEnd", (start, end) => MonthEndIterator(null, start, end));
+        }
+
+        /// <summary>
+        /// Specifies an event should fire on the last tradeable date for the specified
+        /// symbol of each month
+        /// </summary>
+        /// <param name="symbol">The symbol whose exchange is used to determine the last
+        /// tradeable date of the month</param>
+        /// <returns>A date rule that fires on the last tradeable date for the specified security each month</returns>
+        public IDateRule MonthEnd(Symbol symbol)
+        {
+            return new FuncDateRule(symbol.Value + ": MonthEnd", (start, end) => MonthEndIterator(GetSecurity(symbol), start, end));
+        }
+
+        /// <summary>
+        /// Specifies an event should fire on Monday each week
+        /// </summary>
+        /// <returns>A date rule that fires on Monday each week</returns>
+        public IDateRule WeekStart()
```

Comment:
Why wouldn't we just use `Every(DayOfWeek.Monday)` ?

Reply:
Or maybe better leave it as is, to keep the new `WeekStart` event name ?
It's a one-liner in both cases :)

Reply:
Week Start could be tuesday if Monday is a holiday

Reply:
This method is security agnostic, so it will fire every monday. RE: new name -- I dig it :)